chain/src/chain.rs

```diff
@@ -99,6 +99,15 @@ impl ForkChanges {
     pub fn verified_len(&self) -> usize {
         self.attached_blocks.len() - self.dirty_exts.len()
     }
+
+    pub fn is_sorted(&self) -> bool {
```

Comment:
This is inefficient. If it's only used in `debug_assert`, please guard it using `cfg!(debug_assertions)`.

Reply:
Alternative, we can just copy the logic from nightly rust

https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#3205

Reply:
https://github.com/nervosnetwork/ckb/pull/2134/commits/aa8b759209340376efd5f791be43ea7c7888fabb
I bring `rust unstable port` back, introduce `https://github.com/gnzlbg/is_sorted`.